edkrepo_manifest_parser/edk_manifest.py

```diff
@@ -380,6 +396,30 @@ def get_combo_element(self, name):
                 return copy.deepcopy(combo)
         raise ValueError(COMB_UNKOWN_ERROR.format(name, self._fileref))
 
+    def get_all_patchsets(self):
+        return self._patch_sets
+
+    def get_parent_patchset_operations(self, name, patch_set_operations):
+        parentSha = self._patch_sets[name][2]
+        if parentSha in self._patch_sets:
+            if self._patch_sets[parentSha][0] == self._patch_sets[name][0]:
+                self.get_parent_patchset_operations(parentSha, patch_set_operations)
+                patch_set_operations.append(self._patch_set_operations[parentSha])
+            else:
+                raise ValueError(REMOTE_DIFFERENT_ERROR.format(parentSha, self._patch_sets[parentSha][0], 
+                name, self._patch_sets[name][0]))
+
+    def get_patchset_operations(self, name=''):
+        if name != '':

```

Comment:
After implementing the feedback for lines 706-723 This can change to either:
`if name`
or
`if name is not None`